agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -195,10 +195,16 @@ def container_outdated?(service_container)
     # @param [Docker::Container] service_container
     # @return [Boolean]
     def image_outdated?(service_container)
-      image = Docker::Image.get(service_pod.image_name) rescue nil
-      return true unless image
-      return true if image.id != service_container.info['Image']
-
+      begin
+        image = Docker::Image.get(service_pod.image_name)
+        if image.id != service_container.info['Image']
+          info "containers image(#{service_container.info['Image']} does not match images id(#{image.id})}"
+          return true
+        end
+      rescue Docker::Error::NotFoundError
+        info "failed to get image: #{service_pod.image_name}"
```

Comment:
Given, the `ImagePullWorker#ensure_image` will not touch the Docker API if the image is cached... but the `ServicePodWorker` only calls `ImagePullWorker#ensure_image` if the deploy_rev has changed, so it seems guaranteed that there will have been two succesfull calls to `Docker::Image.get` immediately prior to this third call. I seems bizarre if this one would then fail.

So this change to limit the `rescue` => `return true` behavior to the `Docker::Error::NotFound` case is definitely an improvement over the existing `rescue nil', but it doesn't make any sense anymore in the larger picture of how the `ServicePodWorker` and `ImagePullWorker` currently interact?